java/fury-core/src/main/java/org/apache/fury/type/TypeUtils.java

```diff
@@ -737,7 +744,7 @@ private static LinkedHashSet<Class<?>> listBeansRecursiveInclusive(
         TypeRef<?> elementType = getElementType(typeToken);
         LinkedHashSet<TypeRef<?>> newPath = new LinkedHashSet<>(walkedTypePath);
         newPath.add(elementType);
-        beans.addAll(listBeansRecursiveInclusive(elementType.getClass(), newPath, customTypes));
```

Comment:
I think this is a pre-existing bug - `elementType.getClass()` is always `TypeRef.class`, should this be `getRawType` instead?

Reply:
maybe introduced in #2116